### DISCUSSION TITLE
Lesson 12

### DISCUSSION THREAD

**Discussion by itsmohammadh** [Cyfrin/foundry-full-course-cu#1155] [Q&A]
Hi, Can someone explain how the problem that existed here in 4.27 was fixed with the address definition?

```
function mintDsc(uint256 amount, uint256 collateralSeed) public {
        if (usersWithCollateralDeposited.length == 0) {
            return;
        }
        address sender = usersWithCollateralDeposited[collateralSeed % usersWithCollateralDeposited.length];
        (uint256 totalDscMinted, uint256 collateralValueInUsd) = dsce.getAccountInformation(sender);
        int256 maxDscToMint = (int256(collateralValueInUsd) / 2) - int256(totalDscMinted);
        if (maxDscToMint < 0) {
            return;
        }
        amount = bound(amount, 0, uint256(maxDscToMint));
        if (amount == 0) {
            return;
        }
        vm.startPrank(sender);
        dsce.mintDsc(amount);
        vm.stopPrank();
        timeMintIsCalled++;
    }
```

**Comment by 0xusmanf** [comment]
Hey Mate!

Can you please add more context to your question? and what do you mean by 4.27?

**Reply by itsmohammadh** [reply]
I'm sorry, I meant the problem that was in this part of the course at 4:27 minutes (at 4:27 minutes of the video)

**Reply by rocknet** [reply]
Hmm.  4:27 into the video Patrick is talking about DeFi (Uniswap and AAVE).  Can you right click the video at the moment you're talking about, select "Copy video URL at current time" and paste here so we can be on the same page?

**Comment by rocknet** [comment]
I'm not sure what "4.27" is, perhaps a typo of a line number, which we can't see?

I'll take a guess at what you're asking, given you mention the address, you may be asking how adding this line fixed `mintDsc` in the video around 4:15.  
`address sender = usersWithCollateralDeposited[collateralSeed % usersWithCollateralDeposited.length];`

If that's what you're asking, I can explain.  The modulo operator will return the remainder of the integer division of the `collateralSeed` (random) by the length of the `usersWithCollateralDeposited`.  Given it will retain the remainder, this lines up with use as an array index.

For example, say there are three addresses in the array, the remainder of dividing any number by 3 will either be [0, 1, or 2].  This remainder is then used to select one of the addresses that have already deposited to be the user for the rest of the function, so when you `mintDsc`, you at least know you're using someone who has deposited collateral.

Hope that helps.

**Reply by itsmohammadh** [reply]
No, my question is more precisely, how was the bug that existed here fixed by defining the address of the sender? If there is no Sander address, why did it not work with the msg.sender address?

**Reply by rocknet** [reply]
Again, assuming we're talking about 4 hours and 15 minutes into the video (https://youtu.be/wUjYK5gwNZs?t=15302), it "didn't work" before simply because it was never reaching the end of `mintDsc`.  `msg.sender` is a random address, so it's extremely unlikely that a random `msg.sender` would have deposited collateral prior to calling `mintDsc`.  Given that, `mintDsc` would have always returned at line 38.  Hopefully that's what you're asking.  If not, please provide more details.
